codec-http2/src/main/java/io/netty/handler/codec/http2/Http2ConnectionHandler.java

```diff
@@ -60,38 +60,31 @@ public Http2ConnectionHandler(Http2Connection connection, Http2FrameListener lis
     }
 
     public Http2ConnectionHandler(Http2Connection connection, Http2FrameReader frameReader,
-            Http2FrameWriter frameWriter, Http2FrameListener listener) {
-        this(DefaultHttp2ConnectionDecoder.newBuilder().connection(connection)
-                .frameReader(frameReader).listener(listener),
-             DefaultHttp2ConnectionEncoder.newBuilder().connection(connection)
-                .frameWriter(frameWriter));
+                                  Http2FrameWriter frameWriter, Http2FrameListener listener) {
+        checkNotNull(connection, "connection");
```

Comment:
Do we need these null checks here?  Won't `DefaultHttp2ConnectionEncoder` and `DefaultHttp2ConnectionDecoder ` do these?